Getting started Prometheus/ Openmetrics (#4629)
DataDog/documentation: content/en/getting_started/prometheus.md

### Configuration

If running the Agent as a DaemonSet in Kubernetes, configure your Prometheus/Openmetrics check using [auto-discovery](#container-auto-discovery).
If running the Agent as a binary on a host, configure your Prometheus/Openmetrics check as any [other Agent integrations](#host-installation)

#### Host Agent

{{< tabs >}}
{{% tab "Prometheus" %}}

First, edit the `prometheus.d/conf.yaml` file in the `conf.d/` folder at the root of your [Agent's configuration directory][1] to start collecting your Prometheus metrics. See the [sample prometheus.d/conf.yaml][2] for all available configuration options. The minimal configuration of an instance includes:

```yaml
init_config:

instances:

    ## The URL where your application metrics are exposed by Prometheus.
    #
  - prometheus_url: '<PROMETHEUS_METRICS_ENDPOINT>'
    ## The namespace to be appended before all metrics namespace
    #
    namespace: '<METRICS_NAMESPACE_PREFIX_FOR_DATADOG>'

    ## List of `<METRIC_TO_FETCH>: <NEW_METRIC_NAME>` for metrics to be

>     ## fetched from the prometheus endpoint.

Reviewer: ```suggestion
    ## fetched from the Prometheus endpoint.
```

         }
      ]
```

With the following placeholder values:

* `<CONTAINER_NAME>`: name of the container in the pod template section
* `<PROMETHEUS_METRICS_ENDPOINT>`: URL for the metrics as served by the container in Openmetrics format
* `<METRICS_NAMESPACE_PREFIX_FOR_DATADOG>`: set namespace to be prefixed to every metric when viewed in DataDog UI
*  `<OPENMETRICS_METRIC_TO_FETCH>`: Openmetrics metrics key to be fetched from the openmetrics endpoint.
* `<DATADOG_NEW_METRIC_NAME>`: Optional parameter which if set, transforms in Datadog the `<OPENMETRICS_METRIC_TO_FETCH>` metric key to `<DATADOG_NEW_METRIC_NAME>`.

**See the [list of parameters available above](#host-agent) to discover more options for the `ad.datadoghq.com/<CONTAINER_NAME>.instances` annotation.**

[1]: /getting_started/integrations
[2]: /agent/autodiscovery
{{% /tab %}}
{{< /tabs >}}

## From custom to official integration

By default, all metrics retrieved by the generic Prometheus check are considered custom metrics. If you are monitoring off-the-shelf software and think it deserves an official integration, don't hesitate to [contribute][4]!

> Official integrations have their own dedicated directories. There's a default instance mechanism in the generic check to hardcode the default configuration and metrics metadata. For an example, reference the [kube-proxy][5] integration.

Reviewer: ```suggestion
Official integrations have dedicated directories. There's a default instance mechanism in the generic check to hardcode the default configuration and metrics metadata. For example, see the [kube-proxy][5] integration.
```
